src/App.tsx

```diff
@@ -15,48 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+interface SortParms {
+  sortField: string,
+  reverseField: boolean,
+}
+
+function getPrepearedGoods(
+  goods: string[],
+  { sortField, reverseField }: SortParms,
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.alphabet:
+          return good1.localeCompare(good2);
+
+        case SortField.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(false);
+  const visibleGoods = getPrepearedGoods(
+    goodsFromServer, { sortField, reverseField },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(SortField.alphabet)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info',
+            { 'is-light': sortField !== SortField.alphabet })}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortField(SortField.length)}
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success',
+            { 'is-light': sortField !== SortField.length })}
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => setReverseField(!reverseField)}
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning',
+            { 'is-light': !reverseField })}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {
+          (sortField || reverseField) && (
+            <button
+              onClick={() => {
+                setReverseField(false);
+                setSortField('');
+              }}
```

Comment:
```suggestion
              onClick={reset}
```
It's better to move this logic from jsx